lib/elixir/lib/file.ex

```diff
@@ -886,8 +894,6 @@ defmodule File do
     end
   end
 
-  # src may be a file or a directory, dest is definitely
-  # a directory. Returns nil unless an error is found.
   defp do_cp_r(src, dest, callback, acc) when is_list(acc) do
```

Comment:
In this function, we we create a directory, because source has it,
https://github.com/elixir-lang/elixir/pull/8627/files#diff-65687140d0fd6e1c1505681b6fa5f679R911

we are omitting the modes.
I'm also investigating about symbolic link modes, because Macn and Linux behave differently, but all these changes can be part of another Pr